networks/scitile/LAS/conjgrad.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import sys
+import time
+
+import plaidml
+import plaidml.edsl as edsl
+import plaidml.op as op
+import plaidml.exec as plaidml_exec
+from plaidml import Program
+
+RESIDUAL_THRESHOLD = 1e-7
+
+
+def matmul(A, B):
+    I, J = edsl.TensorDims(2)
+    i, j = edsl.TensorIndexes(2)
+    A.bind_dims(I, J)
+    B.bind_dims(J)
+    return edsl.Contraction().outShape(I).outAccess(i).sum(A[i, j] * B[j]).build()
+
+
+def cg_solve(np_A, np_r, timing=False, resthrsh=RESIDUAL_THRESHOLD):
+    N = len(np_r)
+
+    np_x = np.zeros(N)
+    np_rsq = np.matmul(np_r, np_r)
+    np_p = np_r.copy()
+
+    dtype = plaidml.DType.FLOAT32
+
+    A = edsl.Placeholder(dtype, np_A.shape)  #edsl.Tensor(edsl.LogicalShape(dtype, np_A.shape))
```

Comment:
I'm not sure I understand what this comment is telling me.

Reply:
Old comment; should be deleted.